easybuild/easyconfigs/b/BCALM/BCALM-2.2.0-foss-2018a.eb

```diff
@@ -0,0 +1,41 @@
+easyblock = 'CMakeMake'
+
+name = 'BCALM'
+version = '2.2.0'
+
+homepage = 'https://github.com/GATB/bcalm'
+description = """de Bruijn graph compaction in low memory"""
+
+toolchain = {'name': 'foss', 'version': '2018a'}
+
+source_urls = ['https://github.com/GATB/bcalm/archive']
+sources = [
+    'v%(version)s.tar.gz',
+    {
+        'source_urls': ['https://github.com/GATB/gatb-core/archive'],
+        'download_filename': 'da36e81667f46fd34aa274e792d280.tar.gz',
+        'filename': 'gatb-core-20180418.tar.gz',
+        'extract_cmd': 'tar -C bcalm-%(version)s/gatb-core --strip-components=1 -xf %%s'
```

Comment:
@wpoely86 Hmm, it's interesting that this triggers the `TypeError` in your test report.

From a debug log:

```
 == 2018-09-01 16:25:47,510 easyblock.py:378 INFO Added sources: [{...}, {..., 'cmd': 'tar -C bcalm-2.2.0/gatb-core --strip-components=1 -xf %%s', ...}]
```

So, the `%(version)s` gets resolved correctly, but for some reason the escaped `%%s` stays there?!

What happens if you try with just `%s`?

Reply:
OK, so you should just use `%s` here.

The way we resolve templates escapes `%s`, it only tries to template `%(...)s` stuff, and it's smart enough to escape `%s` to `%%s` automatically (and thus `%%s` to `%%%s`):

```
>>> from easybuild.framework.easyconfig.easyconfig import resolve_template
>>> resolve_template("tar -C bcalm-%(version)s/gatb-core --strip-components=1 -xf %%s", {'version': '2.2.0'})
'tar -C bcalm-2.2.0/gatb-core --strip-components=1 -xf %%s'
>>> resolve_template("tar -C bcalm-%(version)s/gatb-core --strip-components=1 -xf %s", {'version': '2.2.0'})
'tar -C bcalm-2.2.0/gatb-core --strip-components=1 -xf %s'
```

I guess we could try and make it even smarter by *not* escaping `%%s` again...